cmd/skaffold/app/cmd/build.go

```diff
@@ -76,21 +76,23 @@ func runBuild(out io.Writer) error {
 
 	bRes, err := createRunnerAndBuildFunc(ctx, buildOut)
 
-	if err != nil {
-		return err
-	}
-
 	if quietFlag {
-		cmdOut := flags.BuildOutput{Builds: bRes}
+		var artifacts []build.Artifact
+		for _, b := range bRes {
+			if b.Error == nil {
+				artifacts = append(artifacts, b.Result)
+			}
```

Comment:
if a one of the artifacts resulted in  an error, we return success now. This might break users potential scripts. 
Shd we collect all the build error in err and then return err?

Reply:
I agree that this is a breaking change that is not great, as build errors will be swallowed - and yes `quietFlag` will be for scripts typically.  

I think here we should actually pass down the whole `[]build.Result` to the `BuildOutput` and have partially failed builds be part of the json output. That would make the errors explicit and scripts can handle it as they wish, CI/CD things can store it as a BuildResult as well...

Reply:
yeah @tejal29 good point. I'll need to be a little smarter about how this is handled.

@balopat I originally agreed with you (and had changed this) but after the `build + deploy = dev` change, I didn't want to change the output because I wasn't sure how `deploy` was expecting to the build output to look. @tejal29 what do you think here? should I leave the templating change to you when you make your change for deploy? or should I just output the full build results, with build target, and either build result or build error?

Reply:
Makes sense! 
1. We can punt having the errors in the output to another PR, just let's not forget about the idea.
2. If we punt, let's just return error on the first error instead of swallowing them

Reply:
sorry just saw this now.  The workflow for CI/CD use case is going to be
```
out =  skaffold build --quiet > results.out
 if $out -ne 1; then
      deploy = skaffold deploy --build-artifacts results.out
endif
```
So, partial deploy of  built artifacts does not make senses.
There is an integration test in place for making sure you pass in the right build output. 
The build output should look like this
https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/flags/build_output_test.go#L47

Reply:
this was actually addressed by adding back the error handling that I (erroneously) removed :)